packages/perseus/src/widgets/interactive-graphs/graphs/utils.ts

```diff
@@ -62,3 +65,49 @@ export function getArrayWithoutDuplicates(array: Array<Coord>): Array<Coord> {
 
     return returnArray;
 }
+
+export function getSlopeStringForLine(line: PairOfPoints, strings): string {
+    const slope = (line[1][1] - line[0][1]) / (line[1][0] - line[0][0]);
+    if (slope === Infinity || slope === -Infinity) {
+        return strings.srLinearGraphSlopeVertical;
+    }
+
+    if (slope === 0) {
+        return strings.srLinearGraphSlopeHorizontal;
+    }
+
+    return slope > 0
+        ? strings.srLinearGraphSlopeIncreasing
+        : strings.srLinearGraphSlopeDecreasing;
+}
+
+export function getInterceptStringForLine(
+    line: PairOfPoints,
+    strings,
+    locale,
+): string {
+    const slope = (line[1][1] - line[0][1]) / (line[1][0] - line[0][0]);
+    const xIntercept = (0 - line[0][1]) / slope + line[0][0];
+    const yIntercept = line[0][1] - slope * line[0][0];
+    const hasXIntercept = xIntercept !== Infinity && xIntercept !== -Infinity;
+    const hasYIntercept = yIntercept !== Infinity && yIntercept !== -Infinity;
```

Comment:
`Number.isFinite` would be a nice simplification here, too. Actually, I think there's a bug that isFinite would fix, because `NaN` _is_ a possibility here. E.g. if `line[0][1]` is 0 and the slope is also 0, the `xIntercept` will be `NaN` (since the line lies on the x-axis). Might be worth proving that with a test.

```suggestion
    const hasXIntercept = Number.isFinite(xIntercept);
    const hasYIntercept = Number.isFinite(yIntercept);
```

Reply:
Oh hmm. You're right that the line crossing the x-axis causes it to be weird. Right now, it says it crosses the x-axis at [some number] comma 0, which isn't really correct.

I'll ask Caitlyn if we should add a custom string for that case, and same thing for Y axis.